### DISCUSSION TITLE
Database root password usage

### DISCUSSION THREAD

**Discussion by Air16** [supabase/supabase#699] [Questions] [answered]
Hello,

For what use cases is it necessary to keep database root password ?

I see at least one case : direct connection using postgres:// url

is there any other ?

Thank you

**Comment by kiwicopple** [comment, marked as answer]
Hey @Air16, good question. A few ideas that you can do with the direct connection

- Connect with a tool like pgadmin/dbeaver/tableplus
- Connect your own middleware servers (for example, a graphql server like Hasura or PostGraphile)
- Dump your database using `pgdump`
- Run migrations - for example if you were using Django/Ecto/sequilize/etc to manage your database tables

Since we are still building, the main reason is to "fill the holes" of functionality which we haven't built yet, or because you have a personal preference for a tool which we don't support

**Reply by Air16** [reply]
Thank you, got it.

And congrats ! Supabase is really amazing, even at this early Beta stage.

All the best to Supabase team and product.
